packages/rendermime/src/outputmodel.ts

```diff
@@ -82,6 +91,16 @@ export class OutputModel implements IOutputModel {
     const { data, metadata, trusted } = Private.getBundleOptions(options);
     this._data = new ObservableJSON({ values: data as JSONObject });
     this._rawData = data;
+    const newData: { [key: string]: any } = {};
```

Comment:
Can you replace `any` here with a proper typing, or worst case scenario an `unknown`?

Reply:
If I replace `any` with `unknown` I get this error:
```
src/outputmodel.ts:103:39 - error TS2322: Type '{ [key: string]: unknown; }' is not assignable to type 'JSONObject'.
  'string' index signatures are incompatible.
    Type 'unknown' is not assignable to type 'JSONValue'.

104     this._data = new ObservableJSON({ values: newData });
                                          ~~~~~~
```

Reply:
That's why `unknown` is useful! Then, I guess it should be:

```suggestion
    const newData: { [key: string]: JSONValue } = {};
```

or `typeof JSONObject` or similar. The use of `any` has lead to countless bugs in JupyterLab which could have been avoided if we used proper typing (and often not because the author who introduced `any` made a mistake, but someone who came after them).

Reply:
But the problem is that for the `'text'` key we want to store an `ObservableString` value, which cannot be assigned to a `JSONValue` or a `PartialJSONValue`.

Reply:
So in that case `_data` is not a valid `ObservableJSON` and the public API in `observableData` will return an invalid value. It seems that you are introducing `observableData` but only ever taking the `text` out of it. Would it make sense to store the `ObservableString` separately and expose it as `observableText` getter instead?

Reply:
> Would it make sense to store the `ObservableString` separately and expose it as `observableText` getter instead?

Good point, I made that change in https://github.com/jupyterlab/jupyterlab/pull/16498/commits/87e04593ca4d2b1c0f028064ba09ba67f9764093.